backend/fms_core/services/experiment_run.py

```diff
@@ -88,13 +91,68 @@ def create_experiment_run(experiment_run_name,
 
     return (experiment_run, errors, warnings)
 
-def launch_experiment_run(pk):
+def start_experiment_run_processing(pk):
+    '''
+    Generates a run info file for an experiment and drops it in a spool directory
+    watched by Tech Dev, which triggers run processing to be scheduled.
+
+    If the run info file is generated without error then the run_processing_launch_date
+    timestamp is updated with the current date and time.
+
+    Args:
+        The ID of the experiment.
+
+    Returns:
+        The experiment run and a list of errors and warnings.
+    '''
     errors = []
     warnings = []
 
-    # For now, just set the launch timestamp on the experiment run...
-    experiment_run = ExperimentRun.objects.get(id=pk)
-    experiment_run.run_processing_launch_date = datetime.now()
-    experiment_run.save()
-   
+    experiment_run = None
+    try:
+        experiment_run = ExperimentRun.objects.get(id=pk)
+    except ExperimentRun.DoesNotExist:
+        errors.append(f'Experiment run with id {pk} not found.')
+
+    if experiment_run is not None:
+        try:
+            generate_run_info_file(experiment_run)
+
+            experiment_run.run_processing_launch_date = timezone.now()
+            experiment_run.save()
+        except Exception as e:
+            errors.append(str(e))
+               
     return (experiment_run, errors, warnings)
+
+def get_run_info_for_experiment(pk):
+    '''
+    Returns a RunInfo object created for the given experiment.
+
+    This function does not modify the experiment's 'launch' status, it simply
+    generates the run info (to be downloaded by the client, for example).
+
+    Args:
+        The ID of the experiment.
+    
+    Returns:
+        RunInfo object, if successful, and a list of errors and warnings.
+    '''
+    run_info = None
+    errors = []
+    warnings = []
+
+    experiment_run = None
+    try:
+        experiment_run = ExperimentRun.objects.get(id=pk)
+    except ExperimentRun.DoesNotExist:
+        errors.append(f'Experiment run with id {pk} not found.')
+
+    if experiment_run is not None:
+        try:
+            run_info = generate_run_info(experiment_run)
+        except Exception as e:
+            errors.append(str(e))
+    
+    return (run_info, errors, [])
```

Comment:
use warnings instead of [] we'll end up setting up a warning that never get returned ...